public/services/CommonService.ts

```diff
@@ -10,15 +10,18 @@ import { IAPICaller } from "../../models/interfaces";
 
 export default class CommonService {
   httpClient: HttpSetup;
+  dataSourceId: string;
 
-  constructor(httpClient: HttpSetup) {
+  constructor(httpClient: HttpSetup, dataSourceId: string = "") {
     this.httpClient = httpClient;
+    this.dataSourceId = dataSourceId;
   }
 
   apiCaller = async <T>(params: IAPICaller): Promise<ServerResponse<T>> => {
     let url = `${NODE_API.API_CALLER}`;
     const payload: HttpFetchOptions = {};
     payload.method = "POST";
+    params.data.dataSourceId = this.dataSourceId;
```

Comment:
We should keep just one convention for passing the dataSourceId. But for now I see three types:
1. In query.dataSourceId
2. In body.dataSourceId
3. In body.data.dataSourceId

Can we try to keep just one?

Reply:
Ack, as this will take some time, will add this to my to-do list.

Reply:
Changed to always pass datasourceid via query param.